connectors/source.py

```diff
@@ -684,6 +689,32 @@ def is_premium():
         """
         return False
 
+    def get_file_extension(self, filename):
+        return get_file_extension(filename)
+
+    def can_file_be_downloaded(self, file_extension, filename, file_size):
+        if file_extension == "":
+            self._logger.debug(
+                f"Files without extension are not supported, skipping {filename}."
+            )
+            return False
+
+        if file_extension.lower() not in TIKA_SUPPORTED_FILETYPES:
+            self._logger.debug(
+                f"Files with the extension {file_extension} are not supported, skipping {filename}."
+            )
+            return False
+
+        if file_size > FILE_SIZE_LIMIT and not self.configuration.get(
```

Comment:
In some cases (Sharepoint REST API) we don't know file sizes, it would be nice to have default handling for it (e.g. not passing file_size always downloads it or something similar)

Reply:
I actually have a draft PR that will allow that! A connector will be able to call the file type or the file size check individually if needed.
https://github.com/elastic/connectors-python/pull/1694

It's dependent on this PR though so I haven't opened it for review yet.